################################################################################
SHORT DESCRIPTION:
################################################################################
Remove the number of cores information exposed via CPUID.

################################################################################
LONG DESCRIPTION:
################################################################################
Supports the option to configure the CPU topology exposed to guests.

################################################################################
CHANGELOG
################################################################################
Derived from former combined patch "hvm-cpuid-multicore"

This patch is also known as "revert-ca2eee92df44.patch" in the XenServer
Xen 4.7 patch queue.

################################################################################
REMOVAL
################################################################################
No

################################################################################
UPSTREAM PLAN
################################################################################
Unknown.

################################################################################
INTERNAL DEPENDENCIES
################################################################################
related to cores-per-socket and hvm-cpuid-multicore

################################################################################
PATCHES
################################################################################
--- a/tools/libxc/xc_cpuid_x86.c
+++ b/tools/libxc/xc_cpuid_x86.c
@@ -324,15 +324,6 @@ static void amd_xc_cpuid_policy(xc_inter
             regs[0] = DEF_MAX_AMDEXT;
         break;
 
-    case 0x80000008:
-        /*
-         * ECX[15:12] is ApicIdCoreSize: ECX[7:0] is NumberOfCores (minus one).
-         * Update to reflect vLAPIC_ID = vCPU_ID * 2.
-         */
-        regs[2] = ((regs[2] + (1u << 12)) & 0xf000u) |
-                  ((regs[2] & 0xffu) << 1) | 1u;
-        break;
-
     case 0x8000000a: {
         if ( !info->nestedhvm )
         {
@@ -372,12 +363,7 @@ static void intel_xc_cpuid_policy(xc_int
     switch ( input[0] )
     {
     case 0x00000004:
-        /*
-         * EAX[31:26] is Maximum Cores Per Package (minus one).
-         * Update to reflect vLAPIC_ID = vCPU_ID * 2.
-         */
-        regs[0] = (((regs[0] & 0x7c000000u) << 1) | 0x04000000u |
-                   (regs[0] & 0x3ffu));
+        regs[0] &= 0x3ffu;
         regs[3] &= 0x3ffu;
         break;
 
@@ -389,11 +375,6 @@ static void intel_xc_cpuid_policy(xc_int
     case 0x80000005:
         regs[0] = regs[1] = regs[2] = 0;
         break;
-
-    case 0x80000008:
-        /* Mask AMD Number of Cores information. */
-        regs[2] = 0;
-        break;
     }
 }
 
@@ -409,15 +390,9 @@ static void xc_cpuid_hvm_policy(xc_inter
         break;
 
     case 0x00000001:
-        /*
-         * EBX[23:16] is Maximum Logical Processors Per Package.
-         * Update to reflect vLAPIC_ID = vCPU_ID * 2.
-         */
-        regs[1] = (regs[1] & 0x0000ffffu) | ((regs[1] & 0x007f0000u) << 1);
-
         regs[2] = info->featureset[featureword_of(X86_FEATURE_SSE3)];
-        regs[3] = (info->featureset[featureword_of(X86_FEATURE_FPU)] |
-                   bitmaskof(X86_FEATURE_HTT));
+        regs[3] = (info->featureset[featureword_of(X86_FEATURE_FPU)] &
+                   ~bitmaskof(X86_FEATURE_HTT));
         break;
 
     case 0x00000007: /* Intel-defined CPU features */
